Gemfile

```diff
@@ -76,7 +76,7 @@ gem "rest-client",                      "~>2.1.0",           :require => false
 gem "ruby_parser",                                           :require => false # Required for i18n string extraction, and DescentdantLoader (via prism)
 gem "ruby-progressbar",                 "~>1.7.0",           :require => false
 gem "rubyzip",                          "~>2.0.0",           :require => false
-gem "rugged",                           "~>1.5.0",           :require => false
+gem "rugged",                           "~>1.9",             :require => false
```

Comment:
Do we have good enough tests for this bump?

It looks like https://github.com/libgit2/rugged/commit/c78ba61e66d7525310e9ede33e809e956812e273 was released in 1.6.2 so wasn't sure if there are good reasons we're upgrading beyond the change that came in 1.6.2.

Reply:
I figured go the latest cause there were some CVEs in libgit2, such as CVE-2024-24577

Coverage wise, I think everything is tested, but I can double check. We don't use a lot out of rugged, and NickL did a lot of the test framework for it. Even so, I saw the warnings on that, so that could only happen if tests exercised it.

Reply:
looks like we use 1.9.0 in the bot: https://github.com/ManageIQ/miq_bot/blob/master/Gemfile.lock#L295

I wasn't sure how well tested the stuff is in git worktree: https://github.com/ManageIQ/manageiq/blob/afd4a7e9acea2a2ae3778c458d5f25e6b893db9d/lib/git_worktree.rb#L196

Reply:
I ran a coverage report:

Running just these 3 specs

```
be rspec spec/models/manageiq/providers/embedded_ansible/automation_manager/configuration_script_source_spec.rb spec/lib/git_worktree_spec.rb spec/models/git_repository_spec.rb
```

I see 92.3% coverage of lib/git_worktree.rb with the only real thing that's not being tested is GitWorktree#checkout. I can add specs for that separately, but since there's code there regarding submodules, it might take a little bit of effort.  Other than that, it's fully covered.

> looks like we use 1.9.0 in the bot:

Yeah part of the reason i updated the bot was to see how bad it was to go to 1.9, because the bot does a lot of the same stuff we do here, and there were no follow up PRs on the bot if I recall.

Reply:
I added a cross-repo test for every repo that uses git stuff.

Reply:
Great, thanks for checking.